src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts

```diff
@@ -299,38 +309,62 @@ export class ExtensionsViewlet extends ViewContainerViewlet implements IExtensio
 				this.defaultRecommendedExtensionsContextKey.set(!this.configurationService.getValue<boolean>(ShowRecommendationsOnlyOnDemandKey));
 			}
 		}, this, this.disposables);
+
+		modes.SuggestRegistry.register({ scheme: 'extensions', pattern: '**/searchinput', hasAccessToAllModels: true }, {
+			triggerCharacters: ['@'],
+			provideCompletionItems: (model: ITextModel, position: Position, _context: modes.SuggestContext) => {
+				return {
+					suggestions: this.autoComplete(model.getValue(), position.column).map(item => (
+						{
+							label: item.fullText,
+							insertText: item.fullText,
+							overwriteBefore: item.overwrite,
+							sortText: item.fullText.indexOf(':') === -1 ? 'a' : item.fullText.indexOf('sort') !== -1 ? 'b' : 'c', // things with no colon; sorts; everything else
+							type: <modes.SuggestionType>'keyword'
+						}))
+				};
+			}
+		});
 	}
 
 	create(parent: HTMLElement): TPromise<void> {
 		addClass(parent, 'extensions-viewlet');
 		this.root = parent;
 
 		const header = append(this.root, $('.header'));
-
-		this.searchBox = append(header, $<HTMLInputElement>('input.search-box'));
-		this.searchBox.placeholder = localize('searchExtensions', "Search Extensions in Marketplace");
-		this.disposables.push(addStandardDisposableListener(this.searchBox, EventType.FOCUS, () => addClass(this.searchBox, 'synthetic-focus')));
-		this.disposables.push(addStandardDisposableListener(this.searchBox, EventType.BLUR, () => removeClass(this.searchBox, 'synthetic-focus')));
+		this.monacoStyleContainer = append(header, $('.monaco-container'));
+		this.searchBox = this.instantiationService.createInstance(CodeEditorWidget, this.monacoStyleContainer, SEARCH_INPUT_OPTIONS, {});
+		this.placeholderText = append(this.monacoStyleContainer, $('.search-placeholder', null, 'Search Extensions in Marketplace'));
 
 		this.extensionsBox = append(this.root, $('.extensions'));
 
-		const onKeyDown = chain(domEvent(this.searchBox, 'keydown'))
-			.map(e => new StandardKeyboardEvent(e));
-		onKeyDown.filter(e => e.keyCode === KeyCode.Escape).on(this.onEscape, this, this.disposables);
```

Comment:
why are removing the binding of esc key to clearing of the text?

Reply:
Looking around on Win10 and MacOS it turns out there are a fair number of places where Esc clears input. But we still don't do in in Code, and browsers `input`s don't do it by default

Reply:
As discussed, lets not try to add this back. esc when suggest widget is open should remove the widget, otherwise esc need not do anything